### PR TITLE
Unit testing to verify payload for sending data

### PR DESCRIPTION
**Pull request by lwilkovich**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by fufranci — COMMENTED** [review]
Can you rebase or merge from master? I think the diff is showing what you have already merged into master.

File: tests/secure_tunneling_tests.c [hunk 1]
```diff
@@ -105,6 +122,55 @@ static void s_send_secure_tunneling_frame_to_websocket(
...
+        AWS_OP_SUCCESS);
+
+    ASSERT_INT_EQUALS(frame_options.payload_length, serialized_st_msg.len + prefix_bytes);
+
+    struct aws_byte_buf out_buf;
+    ASSERT_INT_EQUALS(
+        aws_byte_buf_init(&out_buf, test_context->secure_tunnel->config.allocator, frame_options.payload_length),
+        AWS_OP_SUCCESS);
+
+    ASSERT_INT_EQUALS(secure_tunneling_send_data_call(NULL, &out_buf, frame_options.user_data), true);
```
Comment (fufranci) [inline tests/secure_tunneling_tests.c:153]: ASSERT_TRUE

File: tests/secure_tunneling_tests.c [hunk 2]
```diff
@@ -105,6 +122,55 @@ static void s_send_secure_tunneling_frame_to_websocket(
...
 }
 
+static int s_test_sent_data(
+    struct secure_tunneling_test_context *test_context,
+    const char *expected_payload,
+    const int32_t expected_stream_id,
+    const int prefix_bytes,
+    const enum aws_iot_st_message_type type) {
+
+    struct aws_iot_st_msg c_message;
```
Comment (fufranci) [inline tests/secure_tunneling_tests.c:132]: What does c_message stand for?

Comment (lwilkovich) [reply tests/secure_tunneling_tests.c:182]: just the C version of the pb message

File: source/secure_tunneling.c [hunk 3]
```diff
@@ -280,7 +280,7 @@ static int s_secure_tunneling_close(struct aws_secure_tunnel *secure_tunnel) {
     return AWS_OP_SUCCESS;
 }
 
-static void s_secure_tunneling_on_send_data_complete_callback(
+void s_secure_tunneling_on_send_data_complete_callback(
```
Comment (fufranci) [inline source/secure_tunneling.c:283]: Add back `static`. That is: undo the change on this line.

**Review by fufranci — APPROVED** [review]

**Review by fufranci — DISMISSED** [review]

**Review by fufranci — APPROVED** [review]
